controllers/apps/v1beta3/emqx_handler.go

```diff
@@ -91,75 +88,60 @@ func (r *EmqxReconciler) Do(ctx context.Context, instance appsv1beta3.Emqx) (ctr
 			instance.SetCondition(*condition)
 			_ = r.Status().Update(ctx, instance)
 			return ctrl.Result{RequeueAfter: time.Duration(5) * time.Second}, err
+		} else {
+			condition = appsv1beta3.NewCondition(
+				appsv1beta3.ConditionPluginInitialized,
+				corev1.ConditionTrue,
+				"PluginInitializeSuccessfully",
+				"All default plugins initialized",
+			)
+			instance.SetCondition(*condition)
+			_ = r.Status().Update(ctx, instance)
+			return ctrl.Result{RequeueAfter: time.Duration(5) * time.Second}, nil
 		}
-		condition := appsv1beta3.NewCondition(
-			appsv1beta3.ConditionPluginInitialized,
-			corev1.ConditionTrue,
-			"PluginInitializeSuccessfully",
-			"All default plugins initialized",
-		)
-		instance.SetCondition(*condition)
-		_ = r.Status().Update(ctx, instance)
-		return ctrl.Result{RequeueAfter: time.Duration(5) * time.Second}, nil
 	}
 
-	_, sts = generateLoadedPlugins(instance, sts)
-	_, sts = generateDefaultPluginsConfig(instance, sts)
-
-	acl, sts := generateAcl(instance, sts)
-	resources = append(resources, acl)
+	if acl, _ := generateAcl(instance, sts); acl != nil {
+		resources = append(resources, acl)
+	}
 
-	module, sts := generateLoadedModules(instance, sts)
-	resources = append(resources, module)
+	if module, _ := generateLoadedModules(instance, sts); module != nil {
+		resources = append(resources, module)
+	}
 
-	license, sts := generateLicense(instance, sts)
-	if license != nil {
+	if license, _ := generateLicense(instance, sts); license != nil {
 		resources = append(resources, license)
 	}
 
 	serviceTemplate := instance.GetServiceTemplate()
 	serviceTemplate.MergePorts(r.getListenerPortsByAPI(instance))
 	instance.SetServiceTemplate(serviceTemplate)
-	headlessSvc, svc, sts := generateSvc(instance, sts)
+
+	headlessSvc, svc, _ := generateSvc(instance, sts)
 	resources = append(resources, headlessSvc, svc)
 
-	username, password, apiPort := r.getManagementField(instance)
-	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers, corev1.Container{
-		Name:            "reloader",
-		Image:           "emqx/emqx-operator-reloader:0.0.1",
-		ImagePullPolicy: instance.GetImagePullPolicy(),
-		Env:             sts.Spec.Template.Spec.Containers[0].Env,
-		VolumeMounts:    sts.Spec.Template.Spec.Containers[0].VolumeMounts,
-		Args: []string{
-			"-u", username,
-			"-p", password,
-			"-P", apiPort,
-		},
-	})
+	reloaderContainer := generateReloaderContainer(r, instance, sts)
+	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers, reloaderContainer)
+
+	//add additional containers
+	sts.Spec.Template.Spec.Containers = append(sts.Spec.Template.Spec.Containers, instance.GetAdditionalContainers()...)
+
+	//add additional Annotations
+	setadditionalAnnotations(sts)
```

Comment:
I think the code should be moved to `generateStatefulSetDef`, because now we can look up the emqx container through `findEmqxContainer`